### DISCUSSION TITLE
Redirect stopped working

### DISCUSSION THREAD

**Discussion by Seeker-Smith** [nginx/nginx#420] [Q&A] [answered]
I'm running version 1.24.0 I've been redirecting a PayPal callback in my host server section.

		location = /payment_callback.php?_xfProvider=paypal {
		return 302 https://XXXXXX.com/payment_callback.php?_xfProvider=paypal;
		}
This worked brilliantly up until Dec 26th. Since then redirect has failed. I've tried 301 and 302.

Any help is appreciated.

Thanks,
Peter

**Comment by HanadaLee** [comment]
location do not support args. I believe the following configuration may be more suitable for you. (put it outside any location block)
```
    if ($request_uri = /payment_callback.php?_xfProvider=paypal) {
        return 302 https://XXXXXX.com/payment_callback.php?_xfProvider=paypal;
    }
```

**Reply by Seeker-Smith** [reply]
Thanks for the info, I'll give it a try ad report back.

Regards,
Peter

**Reply by Seeker-Smith** [reply]
Still no joy. The redirect to the new URL is just not working. 

This worked fine until December 18.

location = /payment_callback.php?_xfProvider=paypal {
		return 302 https://xxx-xxx.com/payment_callback.php?_xfProvider=paypal;
		}

This has not worked.

if ($request_uri = /payment_callback.php?_xfProvider=paypal) {
        return 302 https://xxx-xxx.com/payment_callback.php?_xfProvider=paypal;
    }

This has not worked.

if ($request_uri = https://xxx-xxx.com/payment_callback.php?_xfProvider=paypal) {
        return 302 https://xxx-xxx.com/payment_callback.php?_xfProvider=paypal;
    }

**Reply by HanadaLee** [reply]
Can you describe what problem you are having? A simple description of "not working" does not explain the problem you are having. It would be best if you can provide your full configuration.

**Reply by HanadaLee** [reply, marked as answer]
```
    if ($request_uri = /payment_callback.php?_xfProvider=paypal) {
        return 302 https://XXXXXX.com/payment_callback.php?_xfProvider=paypal;
    }
```

In my opinion, if you put this configuration at the beginning of your server block. Reload or restart nginx.

Now request https://yyyy.com/payment_callback.php?_xfProvider=paypal, you should see a 302 redirect with the location header https://XXXXXX.com/payment_callback.php?_xfProvider=paypal

(I assume your domain is yyyy.com)

**Reply by Seeker-Smith** [reply]
Yes, thanks for getting back to me.  

I have moved domains and PayPal is posting back to the old name so I need to redirect the post back to the new name.
 Up until Dec 18 the location style redirect worked. I don't know why but it did. Testing your example did not do a redirect it stayed on the first domain. 

I'm not sure how else I can test this. Could it be because the names are so similar? Or the - in the name?

Old domain sneaker-groups.com
New domain sneaker-forums.com

**Reply by Seeker-Smith** [reply]
Your suggestion to put in the beginning of the server block worked. Thank you so much!!!
